### PR TITLE
The index page now supports an optional 'q' query string param for searching

### PR DESCRIPTION
**Pull request by ryancwalsh**
I love setting up Custom Search Engines in Chrome and Brave browsers, but it seemed impossible to do it for NEAR Explorer until this PR.

chrome://settings/searchEngines

Now, as you can see in this gif, I can set up `ex` (or any key word that I choose) to be a key word that allows me to conveniently search for NEAR accounts or transactions or blocks straight from my Chrome address bar.

custom search engine

### CODE REVIEW

**Comment by ryancwalsh** [comment]
Just following up on this. Do you think this PR will be accepted? I'd use this feature daily and know that people within NF (such as Phillip in FLO team) would too. Thanks.

File: frontend/src/pages/index.tsx [hunk 1]
```diff
@@ -49,7 +50,19 @@ const DashboardContainer = styled(Container, {
...
+  if (q) {
+    return typeof q === "string" ? q : q[0];
+  }
+  return "";
+};
+
 const Dashboard: NextPage = React.memo(() => {
+  const router = useRouter();
+  const query = getQuery(router);
+  console.log({ query });
```
Comment (frol) [inline frontend/src/pages/index.tsx:65]: Let's remove the debug logs

File: frontend/src/components/utils/Search.tsx [hunk 2]
```diff
@@ -259,6 +275,7 @@ const Search: React.FC<Props> = React.memo(({ dashboard }) => {
             autoCapitalize="none"
             onChange={onChange}
             compact={compact}
+            defaultValue={value}
```
Comment (frol) [inline frontend/src/components/utils/Search.tsx:278]: Shouldn't it be `query` instead of `value`?

```suggestion
            defaultValue={query}
```

**Review by frol — APPROVED** [review]
Looks great to me!

**Review by luixo — COMMENTED** [review]
I made some changes to make this feature more idiomatic for the nextjs.
@ryancwalsh please check them out in same-named branch in a original repo: https://github.com/near/near-explorer/tree/feat/support-search-via-query-string
Feel free to cherry pick this commit or comment!

**Comment by ryancwalsh** [comment]
@luixo @frol Great, thanks for all of your work on this. Your version looks good to me. If there is anything you need from me, just let me know. Otherwise, I'll be excited to see this feature soon!

**Comment by luixo** [comment]
> Your version looks good to me.

I'll take it from here then and recreate a PR with my branch.

Thanks for the your efforts!

**Pull request closed without merging**